ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1424,7 +1425,8 @@ int fillFrom(ByteBuf channelReadBuf) {
     void checkAndMaybeComplete() {
       boolean done = false;
       // Now, check if this chunk is done.
-      if (operationTracker.isDone()) {
+      if (operationTracker.isDone() || (chunkException != null
```

Comment:
PutOperation is different from other operations, it doesn't have a `operationCompleted` or `chunkCompleated` boolean variable to terminate operation. Other operations has this variable, so when the quota rejection happens, I can just set this variable to true and the operation would be terminated. But since PutOperation doesn't this variable, I have to check the chunkException and terminate it when quota rejected.